### ISSUE TITLE
More information on the secondary image in the ROFF and RUNW products

### ISSUE THREAD

**Issue by fastice** [isce-framework/nisar-workflows#1]
...->metadata->processingInformation->parameters->secondary->frequency 
includes the following
  azimuthBandwidth
  dopplerCentroid
  rangeBandwidth
  slantRangeSpacing
  zeroDopplerTimeSpacing

My workflow would be greatly simplified if two single value fields could be added. Namely near range (first SLC range) and zeroDopplerTime.  Especially since the spacing is provided, it only makes sense to add the starting values.

The zeroDoppler time can be found in the identification, but I don't see the the secondary starting range.  For consistency, it would be nice to add these values to the corresponding reference data, but for my needs I can back this out from the other slant range data.

**Comment by hfattahi** [comment]
Thanks @fastice . The vectors of ```zeroDopplerTime``` and ```slantRange``` for the ROFF and RUNW products which correspond to the reference image are provided at the following paths in the HDF5 file:
```
/science/LSAR/ROFF/swaths/frequencyA/pixelOffsets/zeroDopplerTime
/science/LSAR/ROFF/swaths/frequencyA/pixelOffsets/slantRange 
```
The first elements of these vectors represent the start time and starting range of the product's radar grid. But looks like you are looking for a starting range of the secondary image to be included in the processing information? Would you need this starting range to be the starting range of the secondary RSLC or do you need this starting to be the range that corresponds to the first common target between the two RSLC images at near range? i.e., do you need the starting range of the secondary RSLC grid or do you need ```reference_starting_range + delta_range``` where ```delta_range``` is the geometrical range offsets between the two images at the starting range of the reference RSLC.

**Comment by fastice** [comment]
Thanks Heresh.  I found everything I need for the reference SLC. In general, I am pretty happy with the HDF products.

I would like to be able to define the R/D coordinates for the secondary SLC, for that I need the reference_starting_range (not there) and the SLC spacing (already included as metadata…->secondary->frequency->slantRangeSpacing). If its just as easy to include the far range too, that would be a nice bonus.

For the azimuth time, I can get this information from  (identification->secondaryZeroDopplerStartTime and …metaData->secondary->frequency->zeroDopplerTimeSpacing), but for consistency, it would be nice to also have the metaData->secondary->frequency->zeroDopplerStartTime (in seconds rather than the full date). 

Another thing that would be nice is to have the secondary’s orbit group, but I can get by if needed with the xml orbit files (though I would vote to have those as hdf products rather than xml). 

One last thing, it would make sense to have ’absoluteOrbitSecondary’ in the identification, which would make it easier to see what the actual pair is (i.e., on someone starts generating skip2 and greater pairs).

To summarize, these are the fields I requesting (best guess for some names) and the * indicates would like but can get by with out.

metadata…->secondary->frequency->nearSlantRange
metadata…->secondary->frequency->farSlantRange *
metadata…->secondary->frequency->zeroDopplerStartTime (seconds) *
metadata…->secondary->frequency->zeroDopplerEndTime (seconds) *

metadata…->secondary->frequency->orbit (same structure as reference orbit and maybe this is not the right place to put it) *

identification->absoluteOrbitSecondary  * 

Ian


Ian Joughin 
Polar Science Center 
Applied Physics Lab 
University of Washington 
1013 NE 40th Street 
Seattle, WA 98105-6698 

Data products:  https://nsidc.org/grimp

Data Tools: https://github.com/fastice/GrIMPTools

Phone: 206-221-3177 
Fax:   206-616-3142

**Comment by hfattahi** [comment]
Thanks for the clarification.

**Comment by fastice** [comment]
Adding @vbrancat

**Comment by vbrancat** [comment]
Hi @fastice,

In our ISCE3 [v0.21.0](https://github.com/isce-framework/isce3/releases/tag/v0.21.0) release, we've added the necessary RSLC parameters to reconstruct the radar grid of the reference and secondary RSLCs. These parameters have been consistently integrated into the full set of NISAR InSAR products (RIFG, RUNW, ROFF, GOFF, GUNW). Please see the attached screenshot showing the processingInformation HDF5 Group of the RIFG sample product shared on the [NISAR website](https://nisar.jpl.nasa.gov/data/sample-data/). Specifically, we've included:

Starting range, slant range spacing, and number of columns for both the reference and secondary RSLCs.
Zero-Doppler start time, zero-Doppler time spacing, and number of lines for both the reference and secondary RSLCs.
With these parameters, you should be able to reconstruct the slant range and zero-Doppler time vectors for the reference and secondary RSLCs.

Additionally, we've added orbit and attitude information for both the reference and secondary RSLCs to all InSAR products. As shown in the attached screenshot, the `orbit` and `attitude` HDF5 Groups are now divided into `reference` and `secondary` groups, each containing the orbit and attitude information for their respective RSLCs.

The only remaining task is to split the `absoluteOrbitNumber` in the identification group into separate datasets for the reference and secondary RSLCs. We will close this issue and open a dedicated issue to address this topic.

![radarGrid_params](https://github.com/user-attachments/assets/568df0c0-3f3f-4cbd-8527-38863247cd27)
![orbit_attitude_params](https://github.com/user-attachments/assets/b11bbcd2-1bed-4bf5-8557-bfac048050b0)
